browser/ui/webui/settings/brave_wallet_handler.cc

```diff
@@ -135,12 +120,72 @@ void BraveWalletHandler::GetCustomNetworksList(
   ResolveJavascriptCallback(args[0], base::Value(json_string));
 }
 
+bool BraveWalletHandler::AddEthereumChainToRPCService(
+    base::Value javascript_callback,
+    const std::string& payload,
+    std::string* error_message) {
+  auto value_to_add = GetEthereumChain(payload, error_message);
+  if (!value_to_add) {
+    *error_message = l10n_util::GetStringUTF8(
+        IDS_SETTINGS_WALLET_NETWORKS_SUMBISSION_FAILED);
+    return false;
+  }
+  absl::optional<brave_wallet::mojom::EthereumChain> chain =
+      brave_wallet::ValueToEthereumChain(value_to_add.value());
+  if (!chain) {
+    *error_message = l10n_util::GetStringUTF8(
+        IDS_SETTINGS_WALLET_NETWORKS_SUMBISSION_FAILED);
+    return false;
+  }
+  PrefService* prefs = Profile::FromWebUI(web_ui())->GetPrefs();
+  if (IsChainExist(prefs, chain->chain_id)) {
+    *error_message =
+        l10n_util::GetStringUTF8(IDS_SETTINGS_WALLET_NETWORKS_EXISTS);
+    return false;
+  }
```

Comment:
I think it makes more sense to move it to `JsonRpcService::AddEthereumChain`, this check should be shared across all possible callers of that API.

Reply:
done